### PR TITLE
[Transforms] Fix type reference directive tests by adding --module amd

### LINKED ISSUE #8182: [Transforms] Missing output for typeReferenceDirectives 11 and 12
Maybe because type references aren't implemented in the new emitter.

Expected: at least one js file to be emitted or at least one error to be created.
Actual: 

```
  15) compiler tests compiler tests for tests/cases/compiler/typeReferenceDirectives11.ts Correct JS output for tests/cases/compiler/typeReferenceDirectives11.ts:
     Error: Expected at least one js file to be emitted or at least one error to be created.
      at Context.<anonymous> (built/local/run.js:80256:31)

  16) compiler tests compiler tests for tests/cases/compiler/typeReferenceDirectives12.ts Correct JS output for tests/cases/compiler/typeReferenceDirectives12.ts:
     Error: Expected at least one js file to be emitted or at least one error to be created.
```

### PR DESCRIPTION
**Pull request by sheetalkamat**
This also reports error when --out is specified and --module isn't specified resulting it to default to commonjs module generation
Fixes #8182

### CODE REVIEW

File: src/compiler/program.ts [hunk 1]
```diff
@@ -2076,7 +2076,10 @@ namespace ts {
             }
 
             // Cannot specify module gen that isn't amd or system with --out
-            if (outFile && options.module && !(options.module === ModuleKind.AMD || options.module === ModuleKind.System)) {
+            // Report this error if user specified --module moduleKind 
+            // or if there is external module in compilation which defaults to commonjs
+            const emitModuleKind = getEmitModuleKind(options);
+            if (outFile && (options.module || firstExternalModuleSourceFile) && !(emitModuleKind === ModuleKind.AMD || emitModuleKind === ModuleKind.System)) {
```
Comment (mhegazy) [inline src/compiler/program.ts]: option.module could be `none`. we should not error in this case.

Comment (sheetalkamat) [inline src/compiler/program.ts]: @mhegazy, really? I think we should.. because --out works with only amd and system.. Nothing else.

Comment (DanielRosenwasser) [inline src/compiler/program.ts]: If you could make it an elaboration I would agree, but right now you end up with two separate error messages that seem confusing.

Comment (mhegazy) [inline src/compiler/program.ts]: works for system, amd, none
errors for node, umd, es6

Comment (sheetalkamat) [inline src/compiler/program.ts]: @mhegazy, That's what I am doing though..  (options.module || firstExternalModuleSourceFile) does exactly same thing. may be I should reword my comment as it seems unclear

Comment (mhegazy) [inline src/compiler/program.ts]: aaah.. `none` is `0`. sorry. for some reason i thought `none` was truthy.

Comment (mhegazy) [inline src/compiler/program.ts]: :+1:

**Pull request merged**